4.1.0 Use PgSql +Jpa  Error accessing table metadata

### Which version of ShardingSphere did you use?
4.1.0
### Which project did you use? Sharding-JDBC or Sharding-Proxy?
Sharding-JDBC
### Expected behavior
startup success
### Actual behavior
startup failed

```
Caused by: org.hibernate.exception.GenericJDBCException: Error accessing table metadata
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:47)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99)
	at org.hibernate.tool.schema.extract.internal.InformationExtractorJdbcDatabaseMetaDataImpl.convertSQLException(InformationExtractorJdbcDatabaseMetaDataImpl.java:98)
	at org.hibernate.tool.schema.extract.internal.InformationExtractorJdbcDatabaseMetaDataImpl.getTables(InformationExtractorJdbcDatabaseMetaDataImpl.java:341)
	at org.hibernate.tool.schema.extract.internal.DatabaseInformationImpl.getTablesInformation(DatabaseInformationImpl.java:120)
	at org.hibernate.tool.schema.internal.GroupedSchemaMigratorImpl.performTablesMigration(GroupedSchemaMigratorImpl.java:65)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.performMigration(AbstractSchemaMigrator.java:207)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.doMigration(AbstractSchemaMigrator.java:114)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:183)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:72)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:310)
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:467)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:939)
	at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:57)
	at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:365)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:390)
	... 20 common frames omitted
Caused by: java.sql.SQLException: Can not find columnLabel TABLE_CAT
	at org.apache.shardingsphere.shardingjdbc.jdbc.core.resultset.DatabaseMetaDataResultSet.findColumn(DatabaseMetaDataResultSet.java:304)
	at org.apache.shardingsphere.shardingjdbc.jdbc.core.resultset.DatabaseMetaDataResultSet.getString(DatabaseMetaDataResultSet.java:147)
	at org.hibernate.tool.schema.extract.internal.InformationExtractorJdbcDatabaseMetaDataImpl.extractTableName(InformationExtractorJdbcDatabaseMetaDataImpl.java:913)
	at org.hibernate.tool.schema.extract.internal.InformationExtractorJdbcDatabaseMetaDataImpl.extractTableInformation(InformationExtractorJdbcDatabaseMetaDataImpl.java:197)
	at org.hibernate.tool.schema.extract.internal.InformationExtractorJdbcDatabaseMetaDataImpl.processTableResults(InformationExtractorJdbcDatabaseMetaDataImpl.java:394)
	at org.hibernate.tool.schema.extract.internal.InformationExtractorJdbcDatabaseMetaDataImpl.getTables(InformationExtractorJdbcDatabaseMetaDataImpl.java:336)
	... 32 common frames omitted
```
### Reason analyze (If you can)

 4.1.0 println log startup failed

```
ShardingSphere-SQL - Logic SQL: select relname from pg_class where relkind='S'
ShardingSphere-SQL - SQLStatement: SelectStatementContext(super=CommonSQLStatementContext(sqlStatement=org.apache.shardingsphere.sql.parser.sql.statement.dml.SelectStatement@7d4069b0, tablesContext=org.apache.shardingsphere.sql.parser.binder.segment.table.TablesContext@73f9fc72), tablesContext=org.apache.shardingsphere.sql.parser.binder.segment.table.TablesContext@73f9fc72, projectionsContext=ProjectionsContext(startIndex=7, stopIndex=13, distinctRow=false, projections=[ColumnProjection(owner=null, name=relname, alias=Optional.empty)]), groupByContext=org.apache.shardingsphere.sql.parser.binder.segment.select.groupby.GroupByContext@6a15ec76, orderByContext=org.apache.shardingsphere.sql.parser.binder.segment.select.orderby.OrderByContext@6e38260f, paginationContext=org.apache.shardingsphere.sql.parser.binder.segment.select.pagination.PaginationContext@10cbdf91, containsSubquery=false)
ShardingSphere-SQL - Actual SQL: db-default ::: select relname from pg_class where relkind='S'
o.h.e.jdbc.spi.SqlExceptionHelper - SQL Error: 0, SQLState: null
o.h.e.jdbc.spi.SqlExceptionHelper - Can not find columnLabel TABLE_CAT
o.s.b.w.s.c.AnnotationConfigServletWebServerApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Invocation of init method failed; nested exception is javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory; nested exception is org.hibernate.exception.GenericJDBCException: Error accessing table metadata
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown completed.
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown completed.
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown completed.
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown completed.
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown completed.
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown completed.
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown completed.
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown initiated...
 com.zaxxer.hikari.HikariDataSource - DatebookHikariCP - Shutdown completed.
```

But 4.0.1  startup success

```
ShardingSphere-SQL - Rule Type: sharding
ShardingSphere-SQL - Logic SQL: select relname from pg_class where relkind='S'
ShardingSphere-SQL - SQLStatement: SelectSQLStatementContext(super=CommonSQLStatementContext(sqlStatement=org.apache.shardingsphere.sql.parser.sql.statement.dml.SelectStatement@675f9bc6, tablesContext=TablesContext(tables=[Table(name=pg_class, alias=Optional.absent())], schema=Optional.absent())), projectionsContext=ProjectionsContext(startIndex=7, stopIndex=13, distinctRow=false, projections=[ColumnProjection(owner=null, name=relname, alias=Optional.absent())], columnLabels=[relname]), groupByContext=org.apache.shardingsphere.sql.parser.relation.segment.select.groupby.GroupByContext@27b8daa1, orderByContext=org.apache.shardingsphere.sql.parser.relation.segment.select.orderby.OrderByContext@b8df260, paginationContext=org.apache.shardingsphere.sql.parser.relation.segment.select.pagination.PaginationContext@4f1a6259, containsSubquery=false)
ShardingSphere-SQL - Actual SQL: db-default ::: select relname from pg_class where relkind='S'
o.s.o.j.LocalContainerEntityManagerFactoryBean - Initialized JPA EntityManagerFactory for persistence unit 'default'
```

--- a/shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/DatabaseMetaDataResultSet.java
+++ b/shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/DatabaseMetaDataResultSet.java
@@ -30,15 +30,7 @@ import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
 import java.sql.Time;
 import java.sql.Timestamp;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
 
 /**
  * Database meta data result set.
@@ -358,10 +350,27 @@ public final class DatabaseMetaDataResultSet extends AbstractUnsupportedDatabase
     @Override
     public int findColumn(final String columnLabel) throws SQLException {
         checkClosed();
-        if (!columnLabelIndexMap.containsKey(columnLabel)) {
-            throw new SQLException(String.format("Can not find columnLabel %s", columnLabel));
+
+        Integer columnIndex = columnLabelIndexMap.get(columnLabel);
+        if (columnIndex != null) {
+            return columnIndex;
         }
-        return columnLabelIndexMap.get(columnLabel);
+
+        columnIndex = columnLabelIndexMap.get(columnLabel.toLowerCase(Locale.US));
+
+        if (columnIndex != null) {
+            columnLabelIndexMap.put(columnLabel, columnIndex);
+            return columnIndex;
+        }
+
+        columnIndex = columnLabelIndexMap.get(columnLabel.toUpperCase(Locale.US));
+
+        if (columnIndex != null) {
+            columnLabelIndexMap.put(columnLabel, columnIndex);
+            return columnIndex;
+        }
+
+        throw new SQLException(String.format("Can not find columnLabel %s", columnLabel));
     }
     
     @Override

--- a/shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/DatabaseMetaDataResultSet.java
+++ b/shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/DatabaseMetaDataResultSet.java
@@ -30,7 +30,15 @@ import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
 import java.sql.Time;
 import java.sql.Timestamp;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
 
 /**
  * Database meta data result set.
@@ -70,7 +78,7 @@ public final class DatabaseMetaDataResultSet extends AbstractUnsupportedDatabase
     }
     
     private Map<String, Integer> initIndexMap() throws SQLException {
-        Map<String, Integer> result = new HashMap<>(resultSetMetaData.getColumnCount());
+        Map<String, Integer> result = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
         for (int i = 1; i <= resultSetMetaData.getColumnCount(); i++) {
             result.put(resultSetMetaData.getColumnLabel(i), i);
         }
@@ -350,27 +358,10 @@ public final class DatabaseMetaDataResultSet extends AbstractUnsupportedDatabase
     @Override
     public int findColumn(final String columnLabel) throws SQLException {
         checkClosed();
-
-        Integer columnIndex = columnLabelIndexMap.get(columnLabel);
-        if (columnIndex != null) {
-            return columnIndex;
+        if (!columnLabelIndexMap.containsKey(columnLabel)) {
+            throw new SQLException(String.format("Can not find columnLabel %s", columnLabel));
         }
-
-        columnIndex = columnLabelIndexMap.get(columnLabel.toLowerCase(Locale.US));
-
-        if (columnIndex != null) {
-            columnLabelIndexMap.put(columnLabel, columnIndex);
-            return columnIndex;
-        }
-
-        columnIndex = columnLabelIndexMap.get(columnLabel.toUpperCase(Locale.US));
-
-        if (columnIndex != null) {
-            columnLabelIndexMap.put(columnLabel, columnIndex);
-            return columnIndex;
-        }
-
-        throw new SQLException(String.format("Can not find columnLabel %s", columnLabel));
+        return columnLabelIndexMap.get(columnLabel);
     }
     
     @Override

--- a/shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/resultset/DatabaseMetaDataResultSetTest.java
+++ b/shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/resultset/DatabaseMetaDataResultSetTest.java
@@ -171,6 +171,15 @@ public final class DatabaseMetaDataResultSetTest {
     }
 
     @Test
+    public void assertGetStringWithLabelCaseInsensitive() throws SQLException {
+        databaseMetaDataResultSet.next();
+        assertThat(databaseMetaDataResultSet.getString(TABLE_NAME_COLUMN_LABEL.toLowerCase()), is(LOGIC_TABLE_NAME));
+        assertThat(databaseMetaDataResultSet.getString(NON_TABLE_NAME_COLUMN_LABEL.toLowerCase()), is("true"));
+        assertThat(databaseMetaDataResultSet.getString(NUMBER_COLUMN_LABEL.toLowerCase()), is("100"));
+        assertThat(databaseMetaDataResultSet.getString(INDEX_NAME_COLUMN_LABEL.toLowerCase()), is(LOGIC_INDEX_NAME));
+    }
+
+    @Test
     public void assertGetNStringWithIndex() throws SQLException {
         databaseMetaDataResultSet.next();
         assertThat(databaseMetaDataResultSet.getNString(1), is(LOGIC_TABLE_NAME));

